lib/utils.js

```diff
@@ -77,16 +77,34 @@ function waitForAvailable(host, port, options, callback) {
       count--;
       if (available) return callback();
       if (count === 0) return callback(new Error('Server is unavailable'));
-      setTimeout(() => run(), options.retryMs);
+      setTimeout(() => run(), options.retryMS);
     });
   }
 
   run();
 }
 
+function createCommandOptions(manager, localOptions) {
+  const options = clone(manager.clientOptions);
+  options.host = manager.options.bind_ip;
+  options.port = manager.options.port;
+  options.connectionTimeout =
+    typeof localOptions.connectionTimeout === 'number' ? localOptions.connectionTimeoutMS : 5000;
+  options.socketTimeout =
+    typeof localOptions.socketTimeout === 'number' ? localOptions.socketTimeoutMS : 0;
+  options.pool = 1;
+
+  // Ensure we only connect once and emit any error caught
+  options.reconnect = false;
+  options.emitError = true;
+
+  return options;
+}
+
 module.exports = {
   clone: clone,
   delay: delay,
   checkAvailable: checkAvailable,
-  waitForAvailable: waitForAvailable
+  waitForAvailable: waitForAvailable,
+  createCommandOptions: createCommandOptions
```

Comment:
nit: we usually change these only when we're working in the area of code they are in, but these can all be exported as a single name:
```
module.exports = {
  clone,
  delay,
  checkAvailable,
  waitForAvailable,
  createCommandOptions
};
```